packages/neon/neon/lib/src/blocs/timer.dart

```diff
@@ -0,0 +1,45 @@
+part of '../../neon.dart';
+
+abstract class TimerBlocEvents {
+  /// Register a [callback] that will be called periodically.
+  /// The time between the executions is defined by the [duration].
+  void registerTimer(final Duration duration, final VoidCallback callback);
+}
+
+abstract class TimerBlocStates {}
+
+/// Execute callbacks at defined periodic intervals.
+/// Components can register their callbacks and everything with the same periodicity will be executed at the same time.
+///
+/// The [TimerBloc] is a singleton.
+/// Sub-second timers are not supported.
+class TimerBloc extends Bloc implements TimerBlocEvents, TimerBlocStates {
+  factory TimerBloc() => _singleton;
+  TimerBloc._internal();
+  static final TimerBloc _singleton = TimerBloc._internal();
+
+  final Map<int, Timer> _timers = {};
+  final Map<int, List<VoidCallback>> _callbacks = {};
+
+  @override
+  void dispose() {
+    for (final timer in _timers.values) {
+      timer.cancel();
+    }
+    super.dispose();
```

Comment:
Why do we need this? isn't `Bloc` just an interface?

Reply:
True, just did it for cleanness (that's not a word lol)